addons/account/report/account_hash_integrity_templates.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ReportAccountHashIntegrity(models.AbstractModel):
+    _name = 'report.account.report_hash_integrity'
+    _description = 'Get hash integrity result as PDF.'
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        return {
+            'doc_ids' : docids,
+            'doc_model' : self.env['res.company'],
+            'data' : data,
+            'docs' : self.env['res.company'].browse(self.env.context.get('active_ids')),
```

Comment:
I don't think you need to put yourself the `docs`/`doc_model`/`doc_ids` values. Maybe I'm wrong but `action_report_account_hash_integrity` define `res.company` as `res_model`. Then, you just need to call `self.env.ref('account.action_report_account_hash_integrity').report_action(self.company_id, results_by_journal)` with a `res.company` record or id as first parameter.